src/main/java/sda/training/rps/app/GameService.java

```diff
@@ -0,0 +1,131 @@
+package sda.training.rps.app;
+
+import sda.training.rps.dao.GameDao;
+import sda.training.rps.dao.IGameDao;
+import sda.training.rps.dao.IStageDao;
+import sda.training.rps.dao.StageDao;
+import sda.training.rps.model.Game;
+import sda.training.rps.model.Player;
+import sda.training.rps.util.Result;
+import sda.training.rps.util.ScannerSingleton;
+
+import java.util.List;
+
+public class GameService implements IGameService {
+    IGameDao gameDao = new GameDao();
+    IStageDao stageDao = new StageDao();
+    Game game = new Game();
+    IStageService stageService = new StageService();
+    ScannerSingleton scanner = ScannerSingleton.getInstance();
+
+    @Override
+    public void startNewGame(Player player) {
+        int winStagesNo = loadStagesProperty();
+        game = createGame(player, winStagesNo);
+        loopStages();
+
+        /*      wczytuje od gracz i ustawia ilosc wygranych etapow
+         *      Twozy instancje gry
+         *      petla stagy
+         * */
+    }
+
+    private int loadStagesProperty() {
+        return 0;
+    }
+
+    private Game createGame(Player player, int winStagesNo) {
+        game.setWinStagesNo(winStagesNo);
+        game.setPlayer(player);
+        //ustawienie poczatku gry
+        return gameDao.mergeObject(game);
+    }
+
+    private void loopStages() {
+        Result result;
+        boolean continueGame = true;
+        while (game.getPlayerScore() < game.getWinStagesNo() ||
+                game.getComputerScore() < game.getWinStagesNo() ||
+                continueGame) {
+            result = stageService.playStage(game);
+            addPoints(result);
+            continueGame = false;
+        }
+        //ustawienie wyniku gry
+        //ustawienie zakonczenia gry (if result win/lose - data zakonczenia)
+        gameDao.mergeObject(game);
+    }
+
+    private void addPoints(Result result) {
+    }
+
+    @Override
+    public void loadOldGame(Player player) {
+        showCustomeMessage();
+        List<Game> games = loadListOfOldGames(player);
+        if (games != null) {
+            games.forEach(System.out::println);
+        } else {
+            System.out.println();
+            return;
+        }
+        int choice = getPlayerChoice(games.size());
+        game = games.get(choice);
+        if (game.getPlayerScore() == 0 &&
+                game.getComputerScore() == 0) {
+            loadScores();
+        }
+        loopStages();
+    }
+
+    private void showCustomeMessage() {
+    }
+
+
+    private List<Game> loadListOfOldGames(Player player) {
+        return null;
```

Comment:
ja wiem ze to zarys ale tam gdzie zwracasz List<> unikaj jak ognia zwracania null. Po to lista ma isEmpty zeby mozna bylo sprawdzic czy jest pusta a zwrocenie null czesto w przyszlosci spowoduje npe (nullPointerException)